php/Blockonomics.php

```diff
@@ -131,59 +131,78 @@ public function check_get_callbacks_response_code($response){
     public function check_get_callbacks_response_body ($response, $crypto){
         $error_str = '';
         $response_body = json_decode(wp_remote_retrieve_body($response));
-
         $callback_secret = get_option('blockonomics_callback_secret');
         $api_url = WC()->api_request_url('WC_Gateway_Blockonomics');
-        $callback_url = add_query_arg('secret', $callback_secret, $api_url);
-        $callback_url_without_schema = preg_replace('/https?:\/\//', '', $callback_url);
-
+        $wordpress_callback_url = add_query_arg('secret', $callback_secret, $api_url);
+        //if merchant doesn't have any xPubs on his Blockonomics account
         if (!isset($response_body) || count($response_body) == 0)
         {
             $error_str = __('You have not entered an xPub', 'blockonomics-bitcoin-payments');
         }
-        elseif (count($response_body) == 1)
+        //if merchant has at least one xPub on his Blockonomics account
+        elseif (count($response_body) >= 1)
         {
-            $response_callback = '';
-            $response_address = '';
-            if(isset($response_body[0])){
-                $response_callback = isset($response_body[0]->callback) ? $response_body[0]->callback : '';
-                $response_address = isset($response_body[0]->address) ? $response_body[0]->address : '';
-            }
-            $response_callback_without_schema = preg_replace('/https?:\/\//', '', $response_callback);
-            if(!$response_callback || $response_callback == null)
-            {
-                //No callback URL set, set one 
-                $this->update_callback($callback_url, $crypto, $response_address);
+            $error_str = $this->examine_server_callback_urls($response_body, $wordpress_callback_url, $api_url, $crypto);
+        }
+        return $error_str;
+    }
+
+    public function examine_server_callback_urls($response_body, $wordpress_callback_url, $api_url, $crypto)
+    {
+        //Scenario 1: Exact match => return $error_str;
```

Comment:
These comments should be unnecessary if the relevant code is commented on correctly and the code understandable. We can add a single comment to `examine_server_callback_urls()` to explain the function checks each existing xpub callback URL to update and/or use.

Reply:
I've reduced the comments into one single line.